### DISCUSSION TITLE
Preloading blocks of images

### DISCUSSION THREAD

**Discussion by Quinlanptq** [jspsych/jsPsych#3132] [Show and tell]
I came up with the following which divides a set of trials into blocks, preloads the images for each block and then presents the blocks.

Might be of interest to those hoping to achieve a similar outcome.

Keen to hear if there is a better way.

Best

Philip.

--------
[scene_check.txt](https://github.com/jspsych/jsPsych/files/12716456/scene_check.txt)

**Comment by Quinlanptq** [comment]
Sorry the preload_b var should read 

    preload_b = {
        type : jsPsychPreload,
        auto_preload: true,
        images: this_block,
        trials : trials_with_variables
    };

**Comment by ChristopheBossens** [comment]
Thanks for sharing!

Btw: did you know that jspsych has a function for shuffling arrays as well: jspsych.randomization.shuffle()? Just pointing this out because I see it quite often that people write their own function, while you can do it out of the box with jsPsych :)

**Comment by Quinlanptq** [comment]
Thanks Christophe - that I didnt know but thank you so much for pointing that out.
